File: php/features.php
<?php
declare(strict_types=1);

// GameOfThronesQuotes SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class GameOfThronesQuotesFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new GameOfThronesQuotesBaseFeature();
            case "test":
                return new GameOfThronesQuotesTestFeature();
            default:
                return new GameOfThronesQuotesBaseFeature();
        }
    }
}

File: py/features.py
# GameOfThronesQuotes SDK feature factory

from feature.base_feature import GameOfThronesQuotesBaseFeature
from feature.test_feature import GameOfThronesQuotesTestFeature


def _make_feature(name):
    features = {
        "base": lambda: GameOfThronesQuotesBaseFeature(),
        "test": lambda: GameOfThronesQuotesTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
